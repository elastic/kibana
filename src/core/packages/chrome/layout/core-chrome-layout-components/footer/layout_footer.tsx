/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the "Elastic License
 * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
 * Public License v 1"; you may not use this file except in compliance with, at
 * your election, the "Elastic License 2.0", the "GNU Affero General Public
 * License v3.0 only", or the "Server Side Public License, v 1".
 */

import React, { ReactNode } from 'react';

import { styles } from './layout_footer.styles';

export interface LayoutFooterProps {
  children: ReactNode;
}

/**
 * The footer slot wrapper
 *
 * @param props - Props for the LayoutFooter component.
 * @returns The rendered LayoutFooter component.
 */
export const LayoutFooter = ({ children }: LayoutFooterProps) => {
  return <footer css={styles.root}>{children}</footer>;
};
