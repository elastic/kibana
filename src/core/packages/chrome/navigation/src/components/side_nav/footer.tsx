/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the "Elastic License
 * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
 * Public License v 1"; you may not use this file except in compliance with, at
 * your election, the "Elastic License 2.0", the "GNU Affero General Public
 * License v3.0 only", or the "Server Side Public License, v 1".
 */

import type { ReactNode } from 'react';
import React, { useRef } from 'react';
import { css } from '@emotion/react';
import { useEuiTheme } from '@elastic/eui';

import { i18n } from '@kbn/i18n';
import { useRovingIndex } from '../../hooks/use_roving_index';

export interface SideNavFooterProps {
  children: ReactNode;
  isCollapsed: boolean;
}

export const SideNavFooter = ({ children, isCollapsed }: SideNavFooterProps): JSX.Element => {
  const ref = useRef<HTMLElement>(null);

  const { euiTheme } = useEuiTheme();

  useRovingIndex(ref);

  return (
    <footer
      aria-label={i18n.translate('core.ui.chrome.sideNavigation.footerAriaLabel', {
        defaultMessage: 'Side navigation footer',
      })}
      css={css`
        align-items: center;
        border-top: 1px solid ${euiTheme.colors.borderBaseSubdued};
        display: flex;
        flex-direction: column;
        gap: ${euiTheme.size.xs};
        justify-content: center;
        padding-top: ${isCollapsed ? euiTheme.size.s : euiTheme.size.m};
      `}
      ref={ref}
    >
      {children}
    </footer>
  );
};
