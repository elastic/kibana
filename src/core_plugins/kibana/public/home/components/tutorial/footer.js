import React from 'react';
import PropTypes from 'prop-types';

import {
  KuiLinkButton,
} from 'ui_framework/components';

import {
  EuiFlexGroup,
  EuiFlexItem,
  EuiHorizontalRule,
  EuiText,
} from '@elastic/eui';

export function Footer({ overviewDashboard }) {
  if (overviewDashboard) {
    return (
      <div>
        <EuiHorizontalRule />

        <EuiFlexGroup justifyContent="spaceBetween" alignItems="flexEnd">

          <EuiFlexItem grow={false}>
            <EuiText>
              <p>
                {`When all steps are complete, you're ready to explore your data.`}
              </p>
            </EuiText>
          </EuiFlexItem>

          <EuiFlexItem
            grow={false}
          >
            <KuiLinkButton
              buttonType="primary"
              href={`#/dashboards?title=${overviewDashboard.title}`}
            >
              {overviewDashboard.linkLabel}
            </KuiLinkButton>
          </EuiFlexItem>

        </EuiFlexGroup>

      </div>
    );
  }
}

Footer.propTypes = {
  overviewDashboard: PropTypes.shape({
    title: PropTypes.string.isRequired,
    linkLabel: PropTypes.string.isRequired,
  })
};
