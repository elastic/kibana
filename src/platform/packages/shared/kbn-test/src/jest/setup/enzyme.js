/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the "Elastic License
 * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
 * Public License v 1"; you may not use this file except in compliance with, at
 * your election, the "Elastic License 2.0", the "GNU Affero General Public
 * License v3.0 only", or the "Server Side Public License, v 1".
 */

import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

/* eslint-env jest */

/**
 * This is a workaround to fix snapshot serialization of emotion classes when rendering React@18 using `import { render } from 'enzyme'`
 * With React@18 emotion uses `useInsertionEffect` to insert styles into the DOM, which enzyme `render` does not trigger because it is using ReactDomServer.renderToString.
 * With React@17 emotion fell back to sync cb, so it was working with enzyme `render`.
 * Without those styles in DOM the custom snapshot serializer is not able to replace the emotion class names.
 * This workaround ensures a fake emotion style tag is present in the DOM before rendering the component with enzyme making the snapshot serializer work.
 */
function mockEnsureEmotionStyleTag() {
  if (!document.head.querySelector('style[data-emotion]')) {
    const style = document.createElement('style');
    style.setAttribute('data-emotion', 'css');
    document.head.appendChild(style);
  }
}

jest.mock('enzyme', () => {
  const actual = jest.requireActual('enzyme');
  return {
    ...actual,
    render: (node, options) => {
      mockEnsureEmotionStyleTag();
      return actual.render(node, options);
    },
  };
});
