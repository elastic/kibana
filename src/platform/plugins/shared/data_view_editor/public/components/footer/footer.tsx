/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the "Elastic License
 * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
 * Public License v 1"; you may not use this file except in compliance with, at
 * your election, the "Elastic License 2.0", the "GNU Affero General Public
 * License v3.0 only", or the "Server Side Public License, v 1".
 */

import React from 'react';
import { i18n } from '@kbn/i18n';

import {
  EuiFlyoutFooter,
  EuiFlexGroup,
  EuiFlexItem,
  EuiButtonEmpty,
  EuiButton,
} from '@elastic/eui';

export enum SubmittingType {
  savingAsAdHoc = 'savingAsAdHoc',
  persisting = 'persisting',
}

interface FooterProps {
  onCancel: () => void;
  onSubmit: (isAdHoc?: boolean) => void;
  onDuplicate?: () => void;
  submittingType: SubmittingType | undefined;
  submitDisabled: boolean;
  hasEditData: boolean;
  isPersisted: boolean;
  allowAdHoc: boolean;
  canSave: boolean;
  isManaged: boolean;
  isDuplicating: boolean;
}

const closeButtonLabel = i18n.translate('indexPatternEditor.editor.flyoutCloseButtonLabel', {
  defaultMessage: 'Close',
});

const saveButtonLabel = i18n.translate('indexPatternEditor.editor.flyoutSaveButtonLabel', {
  defaultMessage: 'Save data view to Kibana',
});

const editButtonLabel = i18n.translate('indexPatternEditor.editor.flyoutEditButtonLabel', {
  defaultMessage: 'Save',
});

const duplicateButtonLabel = i18n.translate(
  'indexPatternEditor.editor.flyoutDuplicateButtonLabel',
  {
    defaultMessage: 'Duplicate',
  }
);

const editUnpersistedButtonLabel = i18n.translate(
  'indexPatternEditor.editor.flyoutEditUnpersistedButtonLabel',
  {
    defaultMessage: 'Continue to use without saving',
  }
);

const exploreButtonLabel = i18n.translate('indexPatternEditor.editor.flyoutExploreButtonLabel', {
  defaultMessage: 'Use without saving',
});

export const Footer = ({
  onCancel,
  onSubmit,
  submittingType,
  submitDisabled,
  hasEditData,
  allowAdHoc,
  isPersisted,
  canSave,
  onDuplicate,
  isManaged,
  isDuplicating,
}: FooterProps) => {
  const isEditingAdHoc = hasEditData && !isPersisted;

  const isEditing = (canSave || isEditingAdHoc) && !isManaged;
  const showDuplicateButton = (canSave || isEditingAdHoc) && onDuplicate;

  const submitPersisted = () => {
    onSubmit(false);
  };
  const submitAdHoc = () => {
    onSubmit(true);
  };

  return (
    <EuiFlyoutFooter>
      <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
        <EuiFlexItem grow={false}>
          <EuiButtonEmpty
            iconType="cross"
            flush="left"
            onClick={onCancel}
            data-test-subj="closeFlyoutButton"
          >
            {closeButtonLabel}
          </EuiButtonEmpty>
        </EuiFlexItem>

        <EuiFlexItem grow={false}>
          <EuiFlexGroup justifyContent="spaceBetween" gutterSize="s" alignItems="center" wrap>
            {showDuplicateButton && (
              <EuiFlexItem grow={false}>
                <EuiButton color="primary" onClick={onDuplicate} data-test-subj="duplicateButton">
                  {duplicateButtonLabel}
                </EuiButton>
              </EuiFlexItem>
            )}

            {allowAdHoc && (
              <EuiFlexItem grow={false}>
                <EuiButton
                  color="primary"
                  onClick={submitAdHoc}
                  data-test-subj="exploreIndexPatternButton"
                  disabled={submitDisabled}
                  isLoading={submittingType === SubmittingType.savingAsAdHoc}
                  title={i18n.translate('indexPatternEditor.editor.flyoutExploreButtonTitle', {
                    defaultMessage: 'Use this data view without creating a saved object',
                  })}
                >
                  {exploreButtonLabel}
                </EuiButton>
              </EuiFlexItem>
            )}

            {isEditing && (
              <EuiFlexItem grow={false}>
                <EuiButton
                  color="primary"
                  onClick={submitPersisted}
                  data-test-subj="saveIndexPatternButton"
                  fill
                  disabled={submitDisabled}
                  isLoading={
                    submittingType === SubmittingType.persisting ||
                    (submittingType === SubmittingType.savingAsAdHoc && isEditingAdHoc)
                  }
                >
                  {hasEditData && !isDuplicating
                    ? isPersisted
                      ? editButtonLabel
                      : editUnpersistedButtonLabel
                    : saveButtonLabel}
                </EuiButton>
              </EuiFlexItem>
            )}
          </EuiFlexGroup>
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiFlyoutFooter>
  );
};
