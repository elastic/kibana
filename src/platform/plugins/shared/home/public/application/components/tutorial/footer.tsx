/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the "Elastic License
 * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
 * Public License v 1"; you may not use this file except in compliance with, at
 * your election, the "Elastic License 2.0", the "GNU Affero General Public
 * License v3.0 only", or the "Server Side Public License, v 1".
 */

import React from 'react';
import { EuiFlexGroup, EuiFlexItem, EuiText, EuiButton } from '@elastic/eui';
import { FormattedMessage } from '@kbn/i18n-react';

interface FooterProps {
  url: string;
  label: string;
}

export function Footer({ url, label }: FooterProps) {
  return (
    <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
      <EuiFlexItem grow={false}>
        <EuiText>
          <p>
            <FormattedMessage
              id="home.exploreYourDataDescription"
              defaultMessage="When all steps are complete, you're ready to explore your data."
            />
          </p>
        </EuiText>
      </EuiFlexItem>

      <EuiFlexItem grow={false}>
        <EuiButton fill href={url}>
          {label}
        </EuiButton>
      </EuiFlexItem>
    </EuiFlexGroup>
  );
}
