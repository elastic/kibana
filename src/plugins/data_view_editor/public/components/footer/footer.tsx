/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { i18n } from '@kbn/i18n';

import {
  EuiFlyoutFooter,
  EuiFlexGroup,
  EuiFlexItem,
  EuiButtonEmpty,
  EuiButton,
} from '@elastic/eui';

interface FooterProps {
  onCancel: () => void;
  onSubmit: () => void;
  submitDisabled: boolean;
}

const closeButtonLabel = i18n.translate('indexPatternEditor.editor.flyoutCloseButtonLabel', {
  defaultMessage: 'Close',
});

const saveButtonLabel = i18n.translate('indexPatternEditor.editor.flyoutSaveButtonLabel', {
  defaultMessage: 'Create data view',
});

export const Footer = ({ onCancel, onSubmit, submitDisabled }: FooterProps) => {
  return (
    <EuiFlyoutFooter className="indexPatternEditor__footer">
      <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
        <EuiFlexItem grow={false}>
          <EuiButtonEmpty
            iconType="cross"
            flush="left"
            onClick={onCancel}
            data-test-subj="closeFlyoutButton"
          >
            {closeButtonLabel}
          </EuiButtonEmpty>
        </EuiFlexItem>

        <EuiFlexItem grow={false}>
          <EuiButton
            color="primary"
            onClick={onSubmit}
            data-test-subj="saveIndexPatternButton"
            fill
            disabled={submitDisabled}
          >
            {saveButtonLabel}
          </EuiButton>
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiFlyoutFooter>
  );
};
