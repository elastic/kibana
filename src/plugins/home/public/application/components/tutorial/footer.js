/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import PropTypes from 'prop-types';

import { EuiFlexGroup, EuiFlexItem, EuiHorizontalRule, EuiText, EuiButton } from '@elastic/eui';

import { FormattedMessage } from '@kbn/i18n/react';

export function Footer({ url, label }) {
  return (
    <div>
      <EuiHorizontalRule />

      <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
        <EuiFlexItem grow={false}>
          <EuiText>
            <p>
              <FormattedMessage
                id="home.exploreYourDataDescription"
                defaultMessage="When all steps are complete, you're ready to explore your data."
              />
            </p>
          </EuiText>
        </EuiFlexItem>

        <EuiFlexItem grow={false}>
          <EuiButton fill href={url}>
            {label}
          </EuiButton>
        </EuiFlexItem>
      </EuiFlexGroup>
    </div>
  );
}

Footer.propTypes = {
  url: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
};
