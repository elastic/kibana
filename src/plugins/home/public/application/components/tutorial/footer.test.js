/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * and the Server Side Public License, v 1; you may not use this file except in
 * compliance with, at your election, the Elastic License or the Server Side
 * Public License, v 1.
 */

import React from 'react';
import { shallow } from 'enzyme';

import { Footer } from './footer';

test('render', () => {
  const component = shallow(<Footer url={'/app/myapp'} label={'launch myapp'} />);
  expect(component).toMatchSnapshot();
});
