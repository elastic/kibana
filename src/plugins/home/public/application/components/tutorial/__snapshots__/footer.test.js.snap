// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<EuiFlexGroup
  alignItems="center"
  justifyContent="spaceBetween"
>
  <EuiFlexItem
    grow={false}
  >
    <EuiText>
      <p>
        <FormattedMessage
          defaultMessage="When all steps are complete, you're ready to explore your data."
          id="home.exploreYourDataDescription"
          values={Object {}}
        />
      </p>
    </EuiText>
  </EuiFlexItem>
  <EuiFlexItem
    grow={false}
  >
    <EuiButton
      color="primary"
      fill={true}
      href="/app/myapp"
      size="m"
    >
      launch myapp
    </EuiButton>
  </EuiFlexItem>
</EuiFlexGroup>
`;
